static/js/message_edit.js

```diff
@@ -16,12 +16,21 @@ var editability_types = {
 };
 exports.editability_types = editability_types;
 
+function is_topic_editable(message, edit_limit_seconds_buffer) {
+    var now = new XDate();
+    edit_limit_seconds_buffer = edit_limit_seconds_buffer || 0;
+
+    // TODO: Change hardcoded value (24 hrs) to be realm setting
+    return message.sent_by_me || (page_params.realm_allow_community_topic_editing
+        && (86400 + edit_limit_seconds_buffer + now.diffSeconds(message.timestamp * 1000) > 0));
+}
+
 function get_editability(message, edit_limit_seconds_buffer) {
     edit_limit_seconds_buffer = edit_limit_seconds_buffer || 0;
     if (!message) {
         return editability_types.NO;
     }
-    if (!(message.sent_by_me || page_params.realm_allow_community_topic_editing)) {
+    if (!is_topic_editable(message, edit_limit_seconds_buffer)) {
```

Comment:
This change is wrong, since it'll block editing by the author if the organization has set a long deadline for that.

I really think the right thing to do is extend the test suite for `get_editibility` carefully for corner cases like this, keep the general ordering of this function (which a  lot of care went into), and then just have other places in the codebase using this call `get_editability`...

Reply:
@shubhamdhama To clear your doubt read [this](https://github.com/zulip/zulip/blob/93ac40105f533e7e815fc1955664e6d2b9e62d2a/static/js/message_edit.js#L51) condition. In case value fo `realm_message_content_edit_limit_seconds` is more than 24hrs, user should be allowed to edit the message but your current code will exit even before reaching the mentioned condition.

Reply:
Oh I think I forgot to address your question regarding the introduction of `is_topic_editable()`. For that I think correct strategy would be to first add unit tests for `get_editability()` function and then extract the `is_topic_editable()` function and use it in other parts of the codebase.

Reply:
>  To clear your doubt read this condition. In case value of realm_message_content_edit_limit_seconds is more than 24hrs, user should be allowed to edit the message but your current code will exit even before reaching the mentioned condition.

Okay, First I'm going to add node tests for this and then extract the function.

Actually I'm still feeling dumb here because I'm the only one who has still didn't get this point,
becuase if `get_editability` function is called for a particular message there can be the following case:
* Message is sent by the user: No matter when message is sent by the user,  `if (!is_topic_editable(message, edit_limit_seconds_buffer)) {` isn't going to return `NO` because `is_topic_editable` function will return true becuase `message.sent_by_me` in 
```js

+    return message.sent_by_me || (page_params.realm_allow_community_topic_editing
+        && (86400 + edit_limit_seconds_buffer + now.diffSeconds(message.timestamp * 1000) > 0));
+}
```
is always true. Hence it will not block any user from editing the message.
 
@HarshitOnGitHub clearing this doubt may solve all my problems here.

Reply:
:man_facepalming: Oops, yeah you are right! I am sorry for any confusion it may have caused to you.